src/Mvc/Mvc.Core/src/ModelBinding/Binders/DateTimeModelBinder.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// An <see cref="IModelBinder"/> for <see cref="DateTime"/> and nullable <see cref="DateTime"/> models.
+    /// </summary>
+    public class DateTimeModelBinder : IModelBinder
+    {
+        private readonly DateTimeStyles _supportedStyles;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DecimalModelBinder"/>.
+        /// </summary>
+        /// <param name="supportedStyles">The <see cref="NumberStyles"/>.</param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+        public DateTimeModelBinder(DateTimeStyles supportedStyles, ILoggerFactory loggerFactory)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _supportedStyles = supportedStyles;
+            _logger = loggerFactory.CreateLogger<DateTimeModelBinder>();
+        }
+
+        /// <inheritdoc />
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            _logger.AttemptingToBindModel(bindingContext);
```

Comment:
I think it makes sense to have this appear before we read the value provider, this log ends up being out of place otherwise:

https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/Mvc.Core/src/ModelBinding/Binders/SimpleTypeModelBinder.cs#L55

I'll update SimpleTypeModelBinder to use this pattern

Reply:
👍